otherlibs/dune-rpc/private/versioned.ml

```diff
@@ -0,0 +1,434 @@
+open Stdune
+open Types
+
+(* This list of methods and generations is exactly the list of existing methods
+   as of commit [3cd240e], which the initial versioning implementation was based
+   on. *)
+let compatibility_menu =
+  Stdune.String.Map.of_list_exn
+    [ ("ping", 1)
+    ; ("diagnostics", 1)
+    ; ("shutdown", 1)
+    ; ("subscribe", 1)
+    ; ("unsubscribe", 1)
+    ; ("build", 1)
+    ; ("status", 1)
+    ; ("notify/abort", 1)
+    ; ("notify/diagnostic", 1)
+    ; ("notify/log", 1)
+    ; ("notify/progress", 1)
+    ]
+
+module Menu = struct
```

Comment:
It's a matter of taste, but i'd move this to its own file.  I see you've already split up dune_rpc_private already - that's good. Let's keep it going and keep Versioned manageable. The Menu itself isn't versioned, so a toplevel name would make more sense.